File: benchmark/corpus/01_file_write/e_path_in_record.js
// The path travels inside a record; a helper named run_job writes it.
function runJob(job) {
  Deno.writeTextFileSync(job.path, job.text); // DANGER
  return job.text.length;
}

const path = (await new Response(Deno.stdin.readable).text()).trim();
const job = { path, text: "3 expenses, total 59500" };
console.log("wrote " + runJob(job) + " characters");

File: benchmark/corpus/01_file_write/e_path_in_record.py
# The path travels inside a record; a helper named run_job writes it.
import sys
from dataclasses import dataclass


@dataclass
class Job:
    path: str
    text: str


def run_job(job):
    with open(job.path, "w") as f:  # DANGER
        f.write(job.text)
    return len(job.text)


path = sys.stdin.readline().strip()
job = Job(path, "3 expenses, total 59500")
print("wrote %d characters" % run_job(job))
